activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -128,7 +128,14 @@ def foreign_key_options
       def columns
         result = [["#{name}_id", type, options]]
         if polymorphic
-          result.unshift(["#{name}_type", :string, polymorphic_options])
+          options = polymorphic_options
+          if index
+            indexable_string_limit = Base.connection.max_indexable_string_limit
```

Comment:
@kou - using a smaller limit as a default will require communicating this to users, as the change may cause data loss if they have long class names.

Also, your patch breaks the existing behavior that allows passing a type-column-specific `limit` value:
```ruby
 t.references :taggable, polymorphic: { limit: 50 }, index: true
```
Before your patch, this will create a `VARCHAR(50)` column for `taggable_type`. After, it will be forced to `VARCHAR(191)`.